packages/expo-video-thumbnails/android/src/main/java/expo/modules/videothumbnails/VideoThumbnailsModule.java

```diff
@@ -0,0 +1,88 @@
+package expo.modules.videothumbnails;
+
+import java.util.Map;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStream;
+import java.io.IOException;
+import java.util.UUID;
+
+import android.content.Context;
+import android.media.MediaMetadataRetriever;
+import android.os.Bundle;
+import android.graphics.Bitmap;
+import android.net.Uri;
+
+import org.unimodules.core.ExportedModule;
+import org.unimodules.core.ModuleRegistry;
+import org.unimodules.core.Promise;
+import org.unimodules.core.interfaces.ExpoMethod;
+import org.unimodules.core.interfaces.ModuleRegistryConsumer;
+import expo.modules.videothumbnails.Options;
+
+public class VideoThumbnailsModule extends ExportedModule implements ModuleRegistryConsumer {
+  private static final String TAG = "ExpoVideoThumbnails";
+  private static final String ERROR_TAG = "E_VIDEO_THUMBNAILS";
+
+  private ModuleRegistry mModuleRegistry;
+
+  public VideoThumbnailsModule(Context context) {
+    super(context);
+  }
+
+  @Override
+  public String getName() {
+    return TAG;
+  }
+
+  @Override
+  public void setModuleRegistry(ModuleRegistry moduleRegistry) {
+    mModuleRegistry = moduleRegistry;
+  }
+
+  private static File ensureDirExists(File dir) throws IOException {
+    if (!(dir.isDirectory() || dir.mkdirs())) {
+      throw new IOException("Couldn't create directory '" + dir + "'");
+    }
+    return dir;
+  }
+
+  private static String generateOutputPath(File internalDirectory, String dirName, String extension) throws IOException {
+    File directory = new File(internalDirectory + File.separator + dirName);
+    ensureDirExists(directory);
+    String filename = UUID.randomUUID().toString();
+    return directory + File.separator + filename + extension;
+  }
+
+  @ExpoMethod
+  public void getThumbnail(String sourceFilename, Map<String, Object> options, final Promise promise) {
+    final Options videoOptions;
+    try {
+      videoOptions = Options.fromArguments(options);
+    } catch (IllegalArgumentException e) {
+      promise.reject(ERROR_TAG + "_INVALID_ARG", e);
+      return;
+    }
+
+    sourceFilename = sourceFilename.replace("file://", "");
+    MediaMetadataRetriever retriever = new MediaMetadataRetriever();
+    retriever.setDataSource(sourceFilename);
+    Bitmap thumbnail = retriever.getFrameAtTime(videoOptions.getTime().intValue()/1000, MediaMetadataRetriever.OPTION_CLOSEST_SYNC);
```

Comment:
Should we call it in some asynchronous task? Can it be a long-running operation?